src/openrct2/paint/TreeContainer.h

```diff
@@ -106,7 +106,8 @@ template<class KeyType, class ValueType> void TreeContainer<KeyType, ValueType>:
         const auto& key = keyValue.first;
         const auto& value = keyValue.second;
 
-        auto keysGenerated = _keyGenerator.GenerateKeys(key);
+        std::vector<KeyType> keysGenerated;
+        _keyGenerator.GenerateKeys(key, keysGenerated);
```

Comment:
This makes no difference performance wise if thats the reason to why you changed it, returning complex objects such as vector is already handled internally as the first parameter, see https://godbolt.org/z/fvYTe57Yq

Reply:
You can gain a bit of performance if you move the vector out of the for loop scope and call clear before calling GenerateKeys, that will avoid allocations as clear will only resize but keeps the capacity.

Reply:
I'm more concerned about the RAM usage currently... If you have any idea on how to reduce that problem.

Reply:
The RAM problem is at the loading, it gets down afterward.

Reply:
[motion_simulator.zip](https://github.com/OpenRCT2/OpenRCT2/files/10495451/motion_simulator.zip)
[twist_paint.zip](https://github.com/OpenRCT2/OpenRCT2/files/10495452/twist_paint.zip)
[enterprise.zip](https://github.com/OpenRCT2/OpenRCT2/files/10495453/enterprise.zip)
Try with these in the object folder.

Reply:
Well as the godbolt demo illustrates changing the function signature like that will have no effect at all besides making it harder to read. I would return the vector if its not supposed to be re-used in any way.